include/boost/url/detail/pct_encoded_view.hpp

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright (c) 2022 Alan Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_DETAIL_PCT_ENCODED_VIEW_HPP
+#define BOOST_URL_DETAIL_PCT_ENCODED_VIEW_HPP
+
+namespace boost {
+namespace urls {
+
+class pct_encoded_view;
+
+namespace detail {
+
+struct unchecked_encoded_view
```

Comment:
oh boy these long identifiers... You can just rename this to `access`. And if we need to add any more functions to it, we can just put them all in here, even if they related to different classes.